Plugins~/Src/MeshSyncClientBlender/msblenModifiers.cpp

```diff
@@ -0,0 +1,327 @@
+#include "msblenModifiers.h"
+#include <sstream>
+#include <MeshSync/SceneGraph/msMesh.h>
+#include "MeshSync/SceneGraph/msPropertyInfo.h"
+#include <msblenUtils.h>
+#include "BlenderPyObjects/BlenderPyContext.h"
+#include "msblenBinder.h"
+
+namespace blender {
+#if BLENDER_VERSION < 300
+	void msblenModifiers::exportProperties(const Object* obj, ms::PropertyManager* propertyManager) {}
+	void msblenModifiers::importProperties(std::vector<ms::PropertyInfo> props) {}
+#else
+
+// Copied from blender source that we cannot include:
+#define IDP_Int(prop) ((prop)->data.val)
+#define IDP_Array(prop) ((prop)->data.pointer)
+#define IDP_Float(prop) (*(float *)&(prop)->data.val)
+#define IDP_Double(prop) (*(double *)&(prop)->data.val)
+#define IDP_String(prop) ((char *)(prop)->data.pointer)
+#define IDP_IDPArray(prop) ((struct IDProperty *)(prop)->data.pointer)
+#define IDP_Id(prop) ((ID *)(prop)->data.pointer)
+
+	std::mutex m_mutex;
+
+	bNodeSocket* getSocketForProperty(IDProperty* property, bNodeTree* group) {
+		for(bNodeSocket* socket : blender::list_range((bNodeSocket*)group->inputs.first)) {
+			if (strcmp(socket->identifier, property->name) == 0) {
+				return socket;
+			}
+		}
+
+		return nullptr;
+	}
+
+	bool doesPropertyUseAttribute(std::string propertyName, NodesModifierData* nodeModifier) {
+		auto attributeName = propertyName + "_use_attribute";
+
+		// Loop through modifier data and get the values:
+		for (auto property : blender::list_range((IDProperty*)nodeModifier->settings.properties->data.group.first)) {
+			if (property->name == attributeName) {
+				return IDP_Int(property);
+			}
+		}
+
+		return false;
+	}
+
+	void addModifierProperties(ModifierData* modifier, const Object* obj, ms::PropertyManager* propertyManager)
+	{
+		if (modifier->type != ModifierType::eModifierType_Nodes) {
+			return;
+		}
+
+		auto nodeModifier = (NodesModifierData*)modifier;
+		auto group = nodeModifier->node_group;
+
+		// Loop through modifier data and get the values:
+		for (auto property : blender::list_range((IDProperty*)nodeModifier->settings.properties->data.group.first)) {
+			if (strstr(property->name, "_use_attribute") || strstr(property->name, "_attribute_name")) {
+				continue;
+			}
+
+			if (doesPropertyUseAttribute(property->name, nodeModifier)) {
+				continue;
+			}
+
+			auto socket = getSocketForProperty(property, group);
+
+			if (socket != nullptr) {
```

Comment:
Can you please invert this to if (socket == nullptr) continue?

Reply:
fixed